zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java

```diff
@@ -481,19 +481,24 @@ protected Object jobRun() throws Throwable {
       try {
         InterpreterContext.set(context);
 
+        InterpreterResult result = null;
+
         // Open the interpreter instance prior to calling interpret().
         // This is necessary because the earliest we can register a hook
         // is from within the open() method.
         LazyOpenInterpreter lazy = (LazyOpenInterpreter) interpreter;
         if (!lazy.isOpen()) {
           lazy.open();
+          result = lazy.executePrecode(context);
```

Comment:
Thanks for explain.

Currently, unittest https://github.com/apache/zeppelin/pull/2221/files#diff-9c5e6fbaa5b23bc0192b4fedc12b2680R445 executes `jdbcInterpreter.executePrecode(interpreterContext);` manually after `open()`. 

Shell we have other unittest that does not call `executePrecode()` manually? so `lazy.executePrecode` in InterpreterServer can be verified.

`zeppelin-interpreter/src/test/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterTest.java` is one place the unittest actually creates RemoteInterpreter process and test against it. This place might be the place you can locate test for this feature.